dpnp/dpnp_iface_indexing.py

```diff
@@ -638,17 +639,16 @@ def fill_diagonal(a, val, wrap=False):
             raise ValueError("All dimensions of input must be of equal length")
         step = sum(a.shape[0] ** x for x in range(a.ndim))
 
-    # TODO: implement flatiter for slice key
-    # a.flat[:end:step] = val
-    fl = a.flat
+    a_sh = a.shape
```

Comment:
please leave a TODO comment, since we would probably need to improve the implementation in the future anyway